browser/brave_content_browser_client.cc

```diff
@@ -236,11 +242,33 @@ void BindBraveSearchHost(
 
   content::BrowserContext* context = render_process_host->GetBrowserContext();
   mojo::MakeSelfOwnedReceiver(
-      std::make_unique<brave_search::BraveSearchHost>(
+      std::make_unique<brave_search::BraveSearchFallbackHost>(
           content::BrowserContext::GetDefaultStoragePartition(context)
               ->GetURLLoaderFactoryForBrowserProcess()),
       std::move(receiver));
 }
+
+void BindBraveSearchDefaultHost(
+    content::RenderFrameHost* const frame_host,
```

Comment:
we shouldn't pass `RenderFrameHost` here directly as the Bind is async it could be invalid at that moment and will crash. We should use `process_id` like here https://github.com/brave/brave-core/pull/8819/files#diff-594631d3a40b383b831cd6c6c401ff278c4b259ec7f1b321808e868ccfb06f2bR236

Reply:
nvm, we are good here as rfh is not bound in `RegisterBrowserInterfaceBindersForFrame`, it comes here